packages/wells/src/client/api/wells.ts

```diff
@@ -81,8 +80,16 @@ export class Wells extends AssetsAPI {
    * ```
    *
    * @param wellName the full name of the well
+   * @param customFilter a custom filter you can apply, input: any -> output: Promise<Well[]>
    */
-  public getWellByName = async (wellName: string): Promise<Well[]> => {
+  public getWellByName = async (
+    wellName: string,
+    customFilter?: SearchWell
+  ): Promise<Well[]> => {
+    if (customFilter) {
+      return await customFilter(wellName);
+    }
```

Comment:
Maybe I'm missing some context here, but why would you want to do this? If you are passing a function and argument to `getWellByName` why not just call the function with the argument yourself?

Reply:
Would it not be better to get the assets by name first and then apply the custom filter? Like in the `getWellsByPolygon` method

Reply:
It was requested to have a method/filter that would totally override the logic of the function as long as it returns `Promise<Well[]>`. AkerBP tenants and BP tentants structure their data differently, so the queries will be different. And yes @drsphelps , I agree you might as well just call the function with the argument yourself. But this is the requested functionally as far as I am concerned.

Reply:
I think we could maybe be a bit more selective about which methods get these custom filters? For example, all the wells no matter how they were ingested will have the CDF id and external id within the asset, so we don't need a custom filter when getting by id?

Reply:
@Sollimann and I had a chat about this earlier today. 

This may not seem not super useful now that we're dealing with wells/wellbores, which are pretty much a given on how to structure. But they are not even the tip of the iceberg. 
When we get to more nuanced datatypes, for a very simple example, well tops (the intersection points of wells with horizons, which are geospatial surfaces) it could be that we model it as, say, as geospatial points in the well data layer, and then some customer decides to model it as events, or as sequences (which is what we currently have in AkerBP, as geospatial was not available in the last hackathon), or even a timeseries if they have time-based data. Then we would want them to be able to override completely, even the type of resource they are looking for, so this custom methods make a lot more sense.

So, for the sake of consistency, if we're going to offer this for the more complex datatypes, we should also offer it to the simpler ones. It doesn't look to me like it's a big deal to add the option to all of the methods, and it reduces the mental overload of trying to decide (and of the user having to check if it is available when they use) which ones have it.
 
Maybe the real issue is if we have too many methods? Should we start by just implementing the ones Discover really uses?

Reply:
Also, @drsphelps , sorry we didn't give you the context before. This is not just related to how data is structured itself, but also to how Discover fetches it. so, the plan was to have this possibility of overriding any of the wells sdk methods (at least the JS one) with another custom method. 
In this way, Discover can always just call the same thing (for example, wells.getByName or, in the future wellbore.listTops(), but have an internal config file that selects which methods to override based on the CDF project/tenant. So BP would have its own list of overriden methods, and other customers could have their own in the future. For any methods that are not overriden, the default well data layer implementation would be called.